Adding support for throwing an Error instead of just returning 'nothing'

allain opened the pull request:
I know that you can change the predicate to throw an exception instead of doing it this way, but it is useful in my case to support changing the guard reaction without changing the predicate.

allain committed:
Adding support for throwing an Error instead of just returning 'nothing'

stoeffel commented:
Nice work :+1: . One thing could you add an example of this behaviour to the `README.md`

allain committed:
Tweaking Readme to show throwing an Error

allain commented:
Sure. Done.

allain committed:
Fixing logic

What I was claiming to check for and what was being checked for were different.

stoeffel commented:
Nice. Thx
